pkg/services/settings/settingsprovider/settingsprovider.go

```diff
@@ -0,0 +1,162 @@
+package settingsprovider
+
+import (
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/services/settings"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+var (
+	logger = log.New("settingsprovider")
+)
+
+func init() {
+	registry.Register(&registry.Descriptor{
+		Name:         "SettingsProvider",
+		Instance:     &Implementation{},
+		InitPriority: registry.High - 5,
+	})
+}
+
+type settingsBag map[string]map[string]string
+
+type Implementation struct {
+	sync.RWMutex
+	FileCfg  *setting.Cfg       `inject:""`
+	SQLStore *sqlstore.SQLStore `inject:""`
+	settings settingsBag
+}
+
+func (i *Implementation) Init() (err error) {
+	i.settings, err = i.loadAndMergeSettings()
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		syncTicker := time.NewTicker(1 * time.Minute)
+		for {
+			<-syncTicker.C
+			logger.Info("Checking for new updates")
+
+			err := i.Refresh()
+			if err != nil {
+				logger.Error("Error while refreshing settings", "error", err.Error())
+			}
+		}
+	}()
+
+	return
+}
+
+func (i *Implementation) Section(name string) settings.Section {
+	i.RLock()
+	defer i.RUnlock()
+
+	keyValues, ok := i.settings[name]
+	if !ok {
+		return section{}
+	}
+
+	return buildSection(keyValues)
+}
+
+func (i *Implementation) KeyValue(section, key string) settings.KeyValue {
+	i.RLock()
+	defer i.RUnlock()
+
+	emptyKV := keyValue{key: key, value: ""}
+
+	_, ok := i.settings[section]
+	if !ok {
+		return emptyKV
+	}
+
+	value, ok := i.settings[section][key]
+	if !ok {
+		return emptyKV
+	}
+
+	return keyValue{key: key, value: value}
+}
+
+func (i *Implementation) Refresh() error {
+	settingsBag, err := i.loadAndMergeSettings()
+	if err != nil {
+		return err
+	}
+
+	var hasBeenUpdated bool
+	i.RLock()
+	if !reflect.DeepEqual(i.settings, settingsBag) {
```

Comment:
I decided to use the `reflect.DeepEqual` function on the initial PoC version for the sake of simplicity. But ideally, it should return a list of changes, that would be sent to the services during the `Reload` in order to let them discern if the reload actually needs to be performed or not.

Any thoughts?? 🤔